File: javascript/packages/flyql/snippets/recipes/active-priority-or-severity.mjs
// Filter active records that are either high priority or critical severity.
//
// FlyQL query: active and (priority = 'high' or severity = 'critical')

import { parse } from 'flyql'
import { generateWhere, newColumn } from 'flyql/generators/clickhouse'

const result = parse("active and (priority = 'high' or severity = 'critical')")

const columns = {
    active: newColumn({ name: 'active', type: 'Bool' }),
    priority: newColumn({ name: 'priority', type: 'String' }),
    severity: newColumn({ name: 'severity', type: 'String' }),
}

const sql = generateWhere(result.root, columns)
console.log(sql)

File: python/snippets/recipes/active_priority_or_severity.py
"""Filter active records that are either high priority or critical severity.

FlyQL query: ``active and (priority = 'high' or severity = 'critical')``
"""

from flyql import parse
from flyql.generators.clickhouse.column import Column
from flyql.generators.clickhouse.generator import to_sql_where

result = parse("active and (priority = 'high' or severity = 'critical')")

columns = {
    "active": Column("active", "Bool"),
    "priority": Column("priority", "String"),
    "severity": Column("severity", "String"),
}

sql = to_sql_where(result.root, columns)
print(sql)
